src/components/dataDisplays/cellRenderers/OverflowController.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { debounce } from 'lodash-es';
+
+import Tooltip from '@material-ui/core/Tooltip';
+
+const noWrapStyles = {
+  display: 'block',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+};
+
+export default function OverflowController({ title, children }) {
+  const [isOverflowed, setIsOverflowed] = useState(false);
+  const [refEl, setRefEl] = useState();
+
+  const ref = useCallback(node => {
+    setRefEl(node);
+  }, []);
+
+  useEffect(
+    () => {
+      if (!refEl) return;
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { clientWidth, scrollWidth } = refEl;
+        const overflow = scrollWidth - clientWidth > 0;
+        setIsOverflowed(overflow);
+      }, 200);
+
+      const resizeObserver = new ResizeObserver(
```

Comment:
if we encounter performance problems with this logic on large tables, we might want to consider a solution that involves one resize observer per column instead of per cell